# Running Graphs
[TOC]

This library contains classes for launching graphs and executing operations.

<a href="../../guide/low_level_intro.md">This guide</a> has examples of how a graph
is launched in a <a href="../../api_docs/python/tf/Session.md"><code>tf.Session</code></a>.

<h2 id="Session_management">Session management</h2>

*   <a href="../../api_docs/python/tf/Session.md"><code>tf.Session</code></a>
*   <a href="../../api_docs/python/tf/InteractiveSession.md"><code>tf.InteractiveSession</code></a>
*   <a href="../../api_docs/python/tf/get_default_session.md"><code>tf.get_default_session</code></a>

<h2 id="Error_classes_and_convenience_functions">Error classes and convenience functions</h2>

*   <a href="../../api_docs/python/tf/OpError.md"><code>tf.OpError</code></a>
*   <a href="../../api_docs/python/tf/errors/CancelledError.md"><code>tf.errors.CancelledError</code></a>
*   <a href="../../api_docs/python/tf/errors/UnknownError.md"><code>tf.errors.UnknownError</code></a>
*   <a href="../../api_docs/python/tf/errors/InvalidArgumentError.md"><code>tf.errors.InvalidArgumentError</code></a>
*   <a href="../../api_docs/python/tf/errors/DeadlineExceededError.md"><code>tf.errors.DeadlineExceededError</code></a>
*   <a href="../../api_docs/python/tf/errors/NotFoundError.md"><code>tf.errors.NotFoundError</code></a>
*   <a href="../../api_docs/python/tf/errors/AlreadyExistsError.md"><code>tf.errors.AlreadyExistsError</code></a>
*   <a href="../../api_docs/python/tf/errors/PermissionDeniedError.md"><code>tf.errors.PermissionDeniedError</code></a>
*   <a href="../../api_docs/python/tf/errors/UnauthenticatedError.md"><code>tf.errors.UnauthenticatedError</code></a>
*   <a href="../../api_docs/python/tf/errors/ResourceExhaustedError.md"><code>tf.errors.ResourceExhaustedError</code></a>
*   <a href="../../api_docs/python/tf/errors/FailedPreconditionError.md"><code>tf.errors.FailedPreconditionError</code></a>
*   <a href="../../api_docs/python/tf/errors/AbortedError.md"><code>tf.errors.AbortedError</code></a>
*   <a href="../../api_docs/python/tf/errors/OutOfRangeError.md"><code>tf.errors.OutOfRangeError</code></a>
*   <a href="../../api_docs/python/tf/errors/UnimplementedError.md"><code>tf.errors.UnimplementedError</code></a>
*   <a href="../../api_docs/python/tf/errors/InternalError.md"><code>tf.errors.InternalError</code></a>
*   <a href="../../api_docs/python/tf/errors/UnavailableError.md"><code>tf.errors.UnavailableError</code></a>
*   <a href="../../api_docs/python/tf/errors/DataLossError.md"><code>tf.errors.DataLossError</code></a>
*   <a href="../../api_docs/python/tf/errors/exception_type_from_error_code.md"><code>tf.errors.exception_type_from_error_code</code></a>
*   <a href="../../api_docs/python/tf/errors/error_code_from_exception_type.md"><code>tf.errors.error_code_from_exception_type</code></a>
*   <a href="../../api_docs/python/tf/errors/raise_exception_on_not_ok_status.md"><code>tf.errors.raise_exception_on_not_ok_status</code></a>
